### ISSUE TITLE
Error for skip links to other pages.

### ISSUE THREAD

**Issue by KeironLowe** [dequelabs/axe-core#2055]
I have some links which are displayed on all pages which link to an anchor on the homepage. These sections are focusable and when I run the scanner on the homepage I don't see any errors, but when I scan a different page they throw the skip links have focusable target error. The links look like `/#about`, `/#services`, `/#contact`.

Expectation: If scanning the page the links go to, then the links should be treated as skip links and scanned as much. If not scanning the page the links go to, then these be ignored.

Actual: The links always throw an error, even if linking to another page

Motivation: Reduce errors in scans

<pre><code>axe-core version: 3.5.1</code></pre>

**Comment by straker** [comment]
So if I understand correctly, you have a few links at the top of the page that  look like:

```html
<a href="/#about"></a>
<a href="/#services"></a>
<a href="/#contact"></a>
```

which fails the skip-link rule for pages that are not the root page (`/`)?

**Comment by KeironLowe** [comment]
@straker yes that's correct.

**Comment by padmavemulapati** [comment]
For three of the above links - givng fail (/#about, /#services, /#contact)
![image](https://user-images.githubusercontent.com/47409579/76215799-79888400-6235-11ea-9fde-bdf5ca553f96.png)
![image](https://user-images.githubusercontent.com/47409579/76215833-8efdae00-6235-11ea-9aa6-d1aeabd9798c.png)
![image](https://user-images.githubusercontent.com/47409579/76215888-a6d53200-6235-11ea-99f9-94e39d88c687.png)

**Comment by straker** [comment]
No docs needed
